### PR TITLE
pythonPackages.woob: init at 3.0

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Motivation for this change

Introduce [woob](https://woob.tech/). Woob is a rebranding of weboob which is already packaged in nixpkgs. After some time, I suggest making "weboob" an alias for "woob".

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 119215` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.woob</li>
    <li>python39Packages.woob</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> After some time, I suggest making "weboob" an alias for "woob".

Python packages do not have aliases yet.

File: pkgs/development/python-modules/woob/default.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    simplejson
+    termcolor
+    unidecode
+  ] ++ lib.optionals isPy27 [ futures ];
+
+  checkPhase = ''
+    nosetests
+  '';
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/woob/default.nix:81]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/python-modules/woob/default.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    homepage = "https://woob.tech";
+    description = "Collection of applications and APIs to interact with websites without requiring the user to open a browser";
+    license = lib.licenses.agpl3;
+    maintainers = [ lib.maintainers.DamienCassou ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/woob/default.nix:85]:
```suggestion
    license = licenses.agpl3;
    maintainers = [ maintainers.DamienCassou ];
```

agpl3Only or agpl3Plus?

File: pkgs/development/python-modules/woob/default.nix [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    # Disable doctests that require networking:
+    sed -i -n -e '/^ *def \+pagination *(.*: *$/ {
+      p; n; p; /"""\|'\'\'\'''/!b
+
+      :loop
+      n; /^ *\(>>>\|\.\.\.\)/ { h; bloop }
+      x; /^ *\(>>>\|\.\.\.\)/bloop; x
+      p; /"""\|'\'\'\'''/b
+      bloop
+    }; p' woob/browser/browsers.py woob/browser/pages.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/woob/default.nix:46]: Can we do a patch for this?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119215` at 7d263a5a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.woob</li>
    <li>python39Packages.woob</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`agpl3` is a deprecated license, please check if project uses `agpl3Plus` or `agpl3Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/woob/default.nix:84:5:
```
   |
84 |     license = lib.licenses.agpl3;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/woob/default.nix (hunk 1 above)
Comment (DamienCassou) [reply pkgs/development/python-modules/woob/default.nix]: fixed

File: pkgs/development/python-modules/woob/default.nix (hunk 2 above)
Comment (DamienCassou) [reply pkgs/development/python-modules/woob/default.nix]: they switched to *L*GPL3 2 years ago: https://gitlab.com/woob/woob/-/blob/master/COPYING

File: pkgs/development/python-modules/woob/default.nix (hunk 3 above)
Comment (DamienCassou) [reply pkgs/development/python-modules/woob/default.nix]: done

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 119215` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.woob</li>
    <li>python39Packages.woob</li>
  </ul>
</details>

File: pkgs/development/python-modules/woob/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/woob/default.nix]: According to https://gitlab.com/woob/woob/-/blob/master/woob/exceptions.py#L10 it is agpl3Plus.

File: pkgs/development/python-modules/woob/default.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
+{ lib, buildPythonPackage, fetchPypi, isPy27
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/woob/default.nix:1]:
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, isPy27
```

Comment (DamienCassou) [reply pkgs/development/python-modules/woob/default.nix]: fixed

File: pkgs/development/python-modules/woob/default.nix (hunk 2 above)
Comment (DamienCassou) [reply pkgs/development/python-modules/woob/default.nix]: you are right. I also confirmed with the authors on IRC.

**Comment by mweinelt** [comment]
```
nix-env failed:
warning: ignoring the user-specified setting 'restrict-eval', because it is a restricted setting and you are not a trusted user
error: while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/.gc-of-borg-outpaths.nix:44:12, called from undefined position:
while evaluating 'g' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/lib/attrsets.nix:294:19, called from undefined position:
while evaluating anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/pkgs/top-level/release-lib.nix:144:38, called from undefined position:
while evaluating 'isDerivation' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/lib/attrsets.nix:323:18, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/pkgs/top-level/release-lib.nix:145:10:
while evaluating 'callPackageWith' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/lib/customisation.nix:117:35, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/pkgs/top-level/python-packages.nix:8377:10:
while evaluating 'makeOverridable' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/lib/customisation.nix:67:24, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/lib/customisation.nix:121:8:
anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/pkgs/development/python-modules/woob/default.nix:1:1 called without required argument 'futures', at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-6/lib/customisation.nix:69:16
``` 

Breaks eval on staging.

**Comment by DamienCassou** [comment]
I'm sorry for the trouble @mweinelt.

**Comment by DamienCassou** [comment]
@SuperSandro2000 Thank you very much for your reviews and for merging.

**Pull request merged**